### PR TITLE
platform/gcp: namespacing resources by cluster_name

### PR DESCRIPTION
**Pull request by enxebre**

### CODE REVIEW

File: modules/gcp/master-igm/master.tf [hunk 1]
```diff
@@ -15,7 +15,7 @@ limitations under the License.
 */
 
 resource "google_compute_instance_template" "tectonic-master-it" {
-  name           = "tectonic-master-it"
+  name           = "${var.cluster_name}-tectonic-master-it"
```
Comment (squat) [inline modules/gcp/master-igm/master.tf]: consider removing `-tectonic` from the name. none of the other names use it.

File: modules/gcp/master-igm/master.tf [hunk 2]
```diff
@@ -49,7 +49,7 @@ resource "google_compute_instance_template" "tectonic-master-it" {
 resource "google_compute_instance_group_manager" "tectonic-master-igm" {
   count              = "${var.instance_count}"
   target_size        = 1
-  name               = "tectonic-master-igm-${count.index}"
+  name               = "${var.cluster_name}-tectonic-master-igm-${count.index}"
```
Comment (squat) [inline modules/gcp/master-igm/master.tf]: same as above.

File: modules/gcp/network/firewall-master.tf [hunk 3]
```diff
@@ -30,7 +30,7 @@ resource "google_compute_firewall" "master-ingress-heapster" {
 }
 
 resource "google_compute_firewall" "master-ingress-flannel" {
-  name    = "master-ingress-flannel"
+  name    = "${var.cluster_name}-master-ingress-flannel"
   network = "${google_compute_network.tectonic-network.name}"
```
Comment (squat) [inline modules/gcp/network/firewall-master.tf]: we should ensure we only create this resource on clusters that actually use Flannel networking.

File: modules/gcp/network/loadbalancer.tf [hunk 4]
```diff
@@ -1,31 +1,43 @@
 resource "google_compute_target_pool" "tectonic-master-targetpool" {
-  name = "tectonic-master-targetpool"
+  name = "${var.cluster_name}-tectonic-master-targetpool"
```
Comment (squat) [inline modules/gcp/network/loadbalancer.tf]: remove `-tectonic` from the name for consistency with the other resources and other providers, e.g. https://github.com/coreos/tectonic-installer/blob/master/modules/aws/master-asg/master.tf

File: modules/gcp/network/loadbalancer.tf [hunk 5]
```diff
@@ -1,31 +1,43 @@
...
 }
 
 resource "google_compute_target_pool" "tectonic-worker-targetpool" {
-  name = "tectonic-worker-targetpool"
+  name = "${var.cluster_name}-tectonic-worker-targetpool"
```
Comment (squat) [inline modules/gcp/network/loadbalancer.tf]: same for this whole file

File: modules/gcp/network/network.tf [hunk 6]
```diff
@@ -16,20 +16,20 @@ limitations under the License.
 
 ## A single GCP network
 resource "google_compute_network" "tectonic-network" {
-  name                    = "tectonic-network"
+  name                    = "${var.cluster_name}-tectonic-network"
```
Comment (squat) [inline modules/gcp/network/network.tf]: same as above for this whole file

File: modules/gcp/worker-igm/worker.tf [hunk 7]
```diff
@@ -15,7 +15,7 @@ limitations under the License.
 */
 
 resource "google_compute_instance_template" "tectonic-worker-it" {
-  name           = "tectonic-worker-it"
+  name           = "${var.cluster_name}-tectonic-worker-it"
```
Comment (squat) [inline modules/gcp/worker-igm/worker.tf]: same for this whole file

File: platforms/gcp/tectonic.tf [hunk 8]
```diff
@@ -13,6 +13,14 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
+locals {
+  etcd_count = "${var.tectonic_experimental ? 0 : max(var.tectonic_etcd_count, 1)}"
```
Comment (squat) [inline platforms/gcp/tectonic.tf]: this is only used once; is it worth creating this local expression? maybe

File: modules/gcp/network/firewall-master.tf (hunk 3 above)
Comment (enxebre) [reply modules/gcp/network/firewall-master.tf]: We'll probably move to use routes and kubenet, will handle this in a follow up networking specific PR

**Comment by enxebre** [comment]
Thanks for having a look @squat, removed 'tectonic' from naming for consistency and pointless 'local'

**Comment by squat** [comment]
ok to test

**Pull request merged**